admin/src/components/EntityLock/index.js

```diff
@@ -21,32 +21,37 @@ export default function EntityLock() {
   const { goBack } = useHistory();
 
   const { formatMessage } = useIntl();
-
+  
   const [isLocked, setIsLocked] = useState(false);
   const [username, setUsername] = useState("");
+  const socket = useRef({});
 
   const { id: userId } = auth.getUserInfo();
+  const lockingData = { entityId: id, entitySlug: slug, userId };
 
-  useEffect(() => {
-    const socket = io(undefined, {
-      reconnectionDelayMax: 10000,
-      rejectUnauthorized: false,
-    });
-
-    const data = { entityId: id, entitySlug: slug, userId };
-
+  const attemptLocking = () => (
```

Comment:
a would add try catch block and throw warning in console when something goes wrong